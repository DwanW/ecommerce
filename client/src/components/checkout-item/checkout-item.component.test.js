import React from 'react';
import { shallow } from 'enzyme';

import {CheckoutItem} from './checkout-item.component';

describe('CheckoutItem component test', ()=> {
    let wrapper;
    let mockAddItem;
    let mockRemoveItem;
    let mockClearItem;

    beforeEach(() => {
        mockAddItem = jest.fn();
        mockRemoveItem = jest.fn();
        mockClearItem = jest.fn();

        const mockProps = {
            cartItem: {
                name:'belt',
                imageUrl: 'www.test.com',
                price: 30,
                quantity: 5
            },
            clearItem: mockClearItem,
            addItem: mockAddItem,
            removeItem: mockRemoveItem
        };

        wrapper = shallow(<CheckoutItem {...mockProps} />);
    });

    it('should render checkoutItem component', ()=> {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call removeItem when it is called',()=> {
        wrapper.find('QuantityContainer').childAt(0).simulate('click');
        expect(mockRemoveItem).toHaveBeenCalled();
    });

    it('should call addItem when it is called',()=> {
        wrapper.find('QuantityContainer').childAt(2).simulate('click');
        expect(mockAddItem).toHaveBeenCalled();
    });

    it('should call clearItem when it is called',()=> {
        wrapper.find('RemoveButtonContainer').simulate('click');
        expect(mockClearItem).toHaveBeenCalled();
    });
});