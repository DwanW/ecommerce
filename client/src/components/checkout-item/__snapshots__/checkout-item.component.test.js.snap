// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component test should render checkoutItem component 1`] = `ShallowWrapper {}`;
